cmd/xl-storage.go

```diff
@@ -2268,18 +2273,12 @@ func (s *xlStorage) writeAllInternal(ctx context.Context, filePath string, b []b
 		return err
 	}
 
-	n, err := w.Write(b)
+	_, err = w.Write(b)
```

Comment:
Why remove the short write check? In what cases was it causing issues?

Reply:
the short write check is never called, file.Write already returns an error in that case

Reply:
I like it as a secondary check, irrespective of the os.File implementation.

Reply:
os.File.Write() method seems to be a generic in all OS. I addressed this differently, let me know what do you think

Reply:
Truncate in all cases then.

I don't really see the point you are trying to make by removing a piece of defensive code from one of the critical write points in MinIO. I don't care if all current code paths make this impossible, but maybe some day it will show up in a stdlib change, and I'd like to handle that gracefully.

Reply:
This change must be reverted @vadmeste

Reply:
it is truncating for all errors now.. yes, I wanted to remove the old code because it confuses debugging but it is better now with your suggestion.

Reply:
no we should never do that check what happens in the EOS branch @vadmeste